paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/ConfirmationDefinition.kt

```diff
@@ -10,97 +10,269 @@ import com.stripe.android.paymentsheet.addresselement.AddressDetails
 import com.stripe.android.paymentsheet.state.PaymentElementLoader
 import kotlinx.parcelize.Parcelize
 
+/**
+ * Defines a confirmation flow that a user might use during confirmation.
+ */
 internal interface ConfirmationDefinition<
     TConfirmationOption : ConfirmationHandler.Option,
     TLauncher,
     TLauncherArgs,
     TLauncherResult : Parcelable
     > {
+    /**
+     * Unique identifier for the definition. Used to order the definitions in same manner when registering launchers.
+     */
     val key: String
 
+    /**
+     * Attempts to cast a given [ConfirmationHandler.Option] to the expected [TConfirmationOption] type of the
+     * implemented definition.
```

Comment:
Should this include an `if successful, it will check if the option can be used for confirmation with canConfirm`?